### DISCUSSION TITLE
Локализация обязательных наборов значений (binding strength = "required").

### DISCUSSION THREAD

**Discussion by Vrodvi** [fhir-ru/core#78] [valueset]
Имеется проблема локализации при подготовке профилей ресурсов для РФ, связанная с "обязательными" наборами значений (binding strength = "`required`").
Такие наборы значений не могут быть заменены на русскоязычные аналоги.
Кроме того, некоторые "обязательные" наборы значений, хотя и подходят по смыслу, но имеют более общее\абстрактное наименование и\или описание, чем это требуется при составлении профиля.

На примере ресурса `Condition`:
- имеется атрибут `clinicalStatus` с "обязательными" набором значений ["Condition Clinical Status Codes"](https://hl7.org/fhir/R4/valueset-condition-clinical.html),
в нём определён концепт "`active`" с наименованием для отображения ("`display`") "`Active`",
который хотелось бы видеть с русским наименование для отображения, и чтобы это наименование было более специфичным для определяемого профиля. 
Например, чтобы в профиле "нозологическая единица" наименование для отображения для этого концепта было "`Активная нозологическая единица`".

---

**Возможные варианты решения:**

**1)** Использовать стандартный набор значений, но в display указывать русское наименование, подходящее для профиля.
ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Активная нозологическая единица"
            }
        ]
    }
```

ЗА: самый простой подход, не требует никаких модификаций набора значений и\или ресурса
ПРОТИВ: противоречит требованиям стандарта: ".. display	Σ	0..1	string	Representation defined by the system"

---

**2)** Подача официального запроса в FHIR на добавление русских наименований и описаний для набора значений (т.е. для стоящей за набором значений системы кодирования), как это сделано для набора значений https://www.hl7.org/fhir/valueset-publication-status.html, чтобы русский перевод стал частью стандарта и его можно было использовать в атрибуте "display", не нарушая требование стандарта. 
Наименования и описания при этом перевести максимально близко к стандарту, а дополнительную специфику профиля описать в руководстве по реализации и не включать в наполнение ресурса.

ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Активное"
            }
        ]
    }
```
, а в руководстве по реализации будет сказано, что в профиле "нозологическая единица" наименование для отображения для этого концепта следует интерпретировать как "Активная нозологическая единица".  

ЗА: перевод станет составной частью стандарта
ПРОТИВ: Организационные сложности при официальном дополнении стандарта. Необходимые более специфические наименования концептов всё равно нельзя будет указать в ресурсе. 

---

**3)** Создание специфического локального справочника с аналогичным набором концептов, но с наименованиями и описаниями на русском языке. 
Использование созданного локального справочника в дополнение к "обязательному", если этого позволяет структура ресурса (конкретно, если тип данных у атрибута = CodeableConcept, который позволяет указание нескольких кодов, обозначающий одно понятие, из разных систем кодирования).

ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Active"
            },
            {
                "code": "active",
                "system": "http://terminology.hl7.ru/CodeSystem/condition-clinical",
                "display": "Активная нозологическая единица"
            }
        ]
    }
```

ЗА: использование стандартных возможностей типа данных CodeableConcept
ПРОТИВ: необходимость дублирования справочника и дублирования данных в ресурсе 

---

**4)** Создание специфического локального справочника с аналогичным набором концептов, но с наименованиями и описаниями на русском языке. 
Создание ресурса `conceptMap` для картирования концептов обязательного набора значений на концепты локализованного справочника.
Использование в ресурсе стандартного набора значений, получение локализованного значения отдельным действием через conceptMap.

ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Active"
            }
        ]
    }
```
, получение локализованного значения через conceptMap: "Active"="Активная нозологическая единица"

ЗА: не требует никаких модификаций набора значений и\или ресурса
ПРОТИВ: необходимость дублирования справочника и дополнительное действие по получению локализованного значения 

---

**5)** Использовать атрибут `text` (для типа данных CodeableConcept).
ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Active"
            }
        ],
        "text": "Активная нозологическая единица"
    }
```

ЗА: использование стандартных возможностей типа данных CodeableConcept
ПРОТИВ: потенциальные проблемы с проверкой соответствия указанного текста и кода, а также сложности для информационных систем с отображением пользователю необходимого набора переведённых значений, т.к. этот набор формально не определён

---

**6)** Использовать расширение с дополнительным набором кодов (по специфическим локальным справочникам) внутри кодируемого элемента.
ПРИМЕР: 
```
    "clinicalStatus": {
        "coding": [
            {
                "code": "active",
                "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
                "display": "Active"
            }
        ],
        "extension": [
            {
                "url": "http://hl7.ru/fhir/StructureDefinition/Сondition-clinicalStatus-ext-codeList",
                "CodeableConcept": [
                    {
                        "code": "active",
                        "system": "http://terminology.hl7.ru/CodeSystem/condition-clinical",
                        "display": "Активная нозологическая единица"
                    }
                ]
            }
        ]        
    }
```

ЗА: расширение можно будет безопасно удалить из ресурса и получить простую базовую структуру, что удобнее для обработки. В отличии от варианта **3)** дополнительное кодирование не будет считаться основным наполнением ресурса. Расширение можно использовать для любого типа данных.
ПРОТИВ: более громоздкая конструкция, возможно более сложное профилирование ресурса.

---

Данная проблема является общей для всех рабочих групп, т.к. все ресурсы имеют атрибуты с обязательными наборами значений на английском языке.
Интересно мнение сообщества.
Предлагается выработать единый принцип действий с обязательными наборами значений.

**Comment by Avazkh** [comment]
В Узбекистане аналогичная проблема, только чуть сложнее - нужна поддержка двух языков (русский и узбекский). Но я смотрел в строну использования CodeSystem.designation.language и designation.value. Не корректно так?

**Reply by Vrodvi** [reply]
Да, для указания переводов на другие языки предназначены атрибуты CodeSystem.designation.language и designation.value, именно так добавлены официальные переводы в стандартный справочник https://www.hl7.org/fhir/valueset-publication-status.html:
```
  {
    "code" : "active",
    "display" : "Active",
    "definition" : "This resource is ready for normal use.",
    "designation" : [{
      "language" : "ru",
      "value" : "активный"
    },
    {
      "language" : "nl",
      "value" : "actief"
    }]
  },
```


Но поскольку это стандартный справочник из спецификации FHIR, то просто добавить в него переводы мы не можем, для этого необходимо подавать заявку на модификацию стандарта - это предложенный вариант "2) Подача официального запроса в FHIR на добавление русских наименований и описаний для набора значений".

Если бы атрибут был связан с набором значений с силой Extensible или Example, то вместо предложенного стандартом справочника можно было бы использовать собственный локальный справочник с любым количеством переводов на разные языки. Проблема как раз с наборами значений, которые жёстко (binding strength = "required") связаны со стандартными справочниками.

Возможно процедура модификации стандартных справочников в спецификации FHIR не такая сложная, и если у участников сообщества имеется такой опыт, то это отличная возможность им поделиться.

**Reply by vadi2** [reply]
Я спросил у наших голландских коллег добавить свой опыт сюда. Мой выбор вторая опция.

Во всех приданных примерах поле `text` не заполнено, почему? Это поле как раз можно переводить:

> The text is the representation of the concept as entered or chosen by the user, and which most closely represents the intended meaning of the user or concept.

```
"coding": [{
        "code": "active",
        "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
        "display": "Active"
}],
"text": "Активная нозологическая единица"
```

**Reply by Vrodvi** [reply]
Тоже думал про вариант с text, но мне кажется такой вариант будет сложно вести и поддерживать.

Указание переведённого значения в атрибуте text без привязки к набору значений - это потенциальные проблемы с проверкой соответствия указанного текста и кода, а также сложности для информационных систем с отображением пользователю необходимого набора переведённых значений, т.к. этот набор формально не определён.

"The text is the representation of the concept" - нет возможности проверить что указанный произвольный текст действительно является представлением этого понятия.

"chosen by the user" - чтобы пользователь выбрал правильное значение в информационной системе его нужно откуда-то взять и без формально определённого набора значений в разных системах этот набор начнёт расходиться.

**Reply by ahenket** [reply]
I don't mean to hijack your discussion but Vadim pointed me here. The way to add translation to a CodeSystem you are not the owner of is by creating a [supplement](https://www.hl7.org/fhir/codesystem.html#supplements).

This would aid a FHIR Terminology Server in offering multilingual capabilities. It is advisable to talk to HL7 Vocabulary WG if you aim to provide these to HL7, e.g. for upload in tx.fhir.org or make them part of [UTG](https://confluence.hl7.org/display/VOC/Unified+Terminology+Governance+Project+%28UTG%29+Page).

**Reply by Vrodvi** [reply]
Alexander, thanks for your suggestion.

So, it is possible to add translations to any codeSystem (that we are not the authors of) and then use these translations as values ​​in display property of resource instances?

For example, we can create a supplement:
```
{
  "resourceType" : "CodeSystem",
  ...
  "url" : "http://terminology.hl7.ru/CodeSystem/condition-clinical",
  "supplements": "http://terminology.hl7.org/CodeSystem/condition-clinical",
  ...
  "content" : "supplement",
  "concept" : [{
    "code" : "active",
	"designation": [
        {
		  "language": "ru",
          "use": {
            "system": "some_system",
            "code": "display"
          },
          "value": "Активная нозологическая единица"
        }
      ]
  },
  ...
  ]
}
```

And then use specified translation in the resource instance:

```
      {
        "code": "active",
        "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
        "display": "Активная нозологическая единица"
      }
```

I'll put this variant in our implementation options list for further discussion.

**Reply by Vrodvi** [reply]
@ahenket 

Alexander,
Could you please comment on another approach to this problem that I found in the Dutch FHIR Implementation Guide - https://simplifier.net/nictizstu3-zib2017/zib-problem.
Additional translations are defined as extensions in this IG.
For example, extension Condition.clinicalStatus.extension:ProblemStatusCodelist is defined for Condition.clinicalStatus.
This extension references valueSet with concepts translated into Dutch - https://simplifier.net/NictizSTU3-Zib2017/2.16.840.1.113883.2.4.3.11.60.40.2.5.1.2--20171231000000/~json.

How relevant is this approach at the moment and is it still used in the Netherlands?

--- 
![image](https://user-images.githubusercontent.com/92928137/153857234-b679bb5a-441c-4f69-9a4b-3183a41a5909.png)
